### PR TITLE
ISPN-6545 Avoid leaking HR server req details

### PR DESCRIPTION
**Pull request by galderz**
https://issues.jboss.org/browse/ISPN-6545

### CODE REVIEW

File: server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodDecoder.java [hunk 1]
```diff
@@ -49,7 +51,7 @@ public NettyTransport getTransport() {
    }
 
    void resetNow() {
-      decodeCtx.resetParams();
```
Comment (wburns) [inline server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodDecoder.java]: We may as well remove this method while we are at it.

**Comment by wburns** [comment]
Changes are fine with me, but these are also the ones I suggested.  If anyone wants to comment please do otherwise I will integrated tomorrow.

**Comment by gustavonalle** [comment]
Changes looks good, but is the node.js client the only way to reproduce it?

**Comment by wburns** [comment]
@gustavonalle unfortunately yes.  It only affected an async clien, where it was sending multiple requests on the same socket.

**Comment by galderz** [comment]
@gustavonalle We'd need an async Java client to replicate indeed.

File: server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodDecoder.java (hunk 1 above)
Comment (galderz) [inline server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodDecoder.java]: Sure

**Comment by galderz** [comment]
@wburns Updated

**Comment by wburns** [comment]
Pulling..

**Comment by wburns** [comment]
Integrated into master, thanks @galderz !

**Pull request closed without merging**